django/core/mail/backends/smtp.py

```diff
@@ -28,25 +31,66 @@ def __init__(
         timeout=None,
         ssl_keyfile=None,
         ssl_certfile=None,
+        *,
+        provider=None,
         **kwargs,
     ):
         super().__init__(fail_silently=fail_silently)
-        self.host = host or settings.EMAIL_HOST
-        self.port = port or settings.EMAIL_PORT
-        self.username = settings.EMAIL_HOST_USER if username is None else username
-        self.password = settings.EMAIL_HOST_PASSWORD if password is None else password
-        self.use_tls = settings.EMAIL_USE_TLS if use_tls is None else use_tls
-        self.use_ssl = settings.EMAIL_USE_SSL if use_ssl is None else use_ssl
-        self.timeout = settings.EMAIL_TIMEOUT if timeout is None else timeout
-        self.ssl_keyfile = (
-            settings.EMAIL_SSL_KEYFILE if ssl_keyfile is None else ssl_keyfile
-        )
-        self.ssl_certfile = (
-            settings.EMAIL_SSL_CERTFILE if ssl_certfile is None else ssl_certfile
-        )
+
+        if provider is not None:
+            # Being initialized from EMAIL_PROVIDERS.
+            options = settings.EMAIL_PROVIDERS[provider].get("OPTIONS", {})
+            self.host = host or options.get("host", self.DEFAULT_HOST)
+            self.port = port or options.get("port", self.DEFAULT_PORT)
+            self.username = options.get("username") if username is None else username
+            self.password = options.get("password") if password is None else password
+            self.use_tls = options.get("use_tls") if use_tls is None else use_tls
+            self.use_ssl = options.get("use_ssl") if use_ssl is None else use_ssl
+            self.ssl_keyfile = (
+                options.get("ssl_keyfile") if ssl_keyfile is None else ssl_keyfile
+            )
+            self.ssl_certfile = (
+                options.get("ssl_certfile") if ssl_certfile is None else ssl_certfile
+            )
+            self.timeout = options.get("timeout") if timeout is None else timeout
```

Comment:
An EmailBackend **must not** examine settings.EMAIL_PROVIDERS directly. (That is mail.get_connection()'s responsibility.)

```suggestion
        if provider is not None:
            # Being initialized from EMAIL_PROVIDERS.
            self.host = self.DEFAULT_HOST if host is None else host
            self.port = self.DEFAULT_PORT if port is None else port
            self.username = username
            self.password = password
            self.use_tls = False if (use_tls is None) else use_tls
            self.use_ssl = False if (use_ssl is None) else use_ssl
            self.ssl_keyfile = ssl_keyfile
            self.ssl_certfile = ssl_certfile
            self.timeout = timeout
```